tests/get_external_dependencies.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python3
+from __future__ import annotations
+
+import os
+import sys
+
+from utils import read_dependencies
+
+distributions = sys.argv[1:]
+if not distributions:
+    distributions = os.listdir("stubs")
+
+for distribution in distributions:
+    for package in read_dependencies(distribution).external_pkgs:
+        print(package)
```

Comment:
This will be problematic if we ever have packages that depend on, e.g., different versions of numpy. But that definitely feels like a hypothetical problem that we can tackle if and when it comes up -- this is good enough for now!

We _could_ consider doing some deduplication here, e.g.:

```suggestion
requirements: set[str] = set()

for distribution in distributions:
    requirements.update(read_dependencies(distribution).external_pkgs)

for requirement in sorted(requirements):
    print(requirement)
```

Reply:
>  e.g., different versions of numpy

Only if their version ranges are not compatible. And we're only affected by types, not implementation. So should be extremely rare imo, but yeah. I think the only alternative is a virtual environment. And I like this currently very simple solution 😄